chain/ethereum/src/adapter.rs

```diff
@@ -445,22 +445,11 @@ impl EthereumCallFilter {
             return false;
         }
 
-        // Make sure the call input size is multiple of 32, otherwise we can't decode it.
-        // This is due to the Ethereum ABI spec: https://docs.soliditylang.org/en/v0.8.11/abi-spec.html
+        // We don't have to check whether the call.input.len() is multiple of
+        // 32 because we now skip invalid decoding of calls.
         //
-        // A scenario where the input could have the wrong size is when a `delegatecall` is
-        // "disguised" as a `call`. For example for this couple of transactions/calls:
-        // 1. https://etherscan.io/tx/0x8e992eeb40e18703dd8169a8031e2113311985f1c20e0723a2dc362df40bafb0
-        // 2. https://etherscan.io/tx/0x7c391bcfa2007f84e123ad47ea72e2d6ffb2fbab81deff0990b0c499e0664a92
-        //
-        // The first one is acting as a proxy to the second one (an atomicMatch_).
-        // If you try to decode the first call as it is/comes from a `traces` RPC request, it
-        // will fail because it has a smaller size/length.
-        let correct_input_size = (call.input.0.len() - 4) % 32 == 0;
-        if !correct_input_size {
-            return false;
-        }
-
+        // - https://github.com/graphprotocol/graph-node/pull/3326
+        // - ae6951328df4270e3153897220fdef1822055859
```

Comment:
This comment is hard to understand if you don't have all the context. Maybe this should say that decoding failures are filtered elsewhere, and have a 'see also' to that code.

Reply:
Ok, I'll improve on it!

Reply:
Do you think it's better now or do you have some other suggestion? 📝